### PR TITLE
feat: allow borsh-serialized contract fn arguments

### PR DESCRIPTION
**Pull request by chadoh**
Let's say you have a `contract` object instantiated in your code, and the contract it wraps has an `add_person` function which receives a borsh-serialized struct. The following code now works:

    import { serialize } from 'near-api-js/lib/utils/serialize'

    class Person {
      constructor (proof) {
        Object.assign(this, proof)
      }
    }

    const personSchema = new Map([
      [Person, {
        kind: 'struct',
        fields: [
          ['age', 'u64'],
          ['name', 'String'],
        ]
      }]
    ])

    const person = new Person({ age: 33, name: 'Jesus' })

    contract.add_person(serialize(personSchema, person))

### CODE REVIEW

File: src/contract.ts [hunk 1]
```diff
@@ -40,7 +47,10 @@ export class Contract {
                 writable: false,
                 enumerable: true,
                 value: nameFunction(methodName, async (args: object = {}, gas?: BN, amount?: BN, ...ignored) => {
-                    if (ignored.length || Object.prototype.toString.call(args) !== '[object Object]') {
+                    if (
```
Comment (vgrichina) [inline src/contract.ts:50]: nit: this formatting of `if (` almost gives me physical pain lol

**Review by vgrichina — APPROVED** [review]

**Comment by vgrichina** [comment]
@chadoh need to fix test failures

File: src/contract.ts (hunk 1 above)
Comment (chadoh) [reply src/contract.ts:50]: hahaha, ok, I'll find a way to shorten it enough to fit on one line comfortably

**Comment by chadoh** [comment]
@vgrichina merged; could you cut a release?

**Pull request merged**